chemprop/cli/fingerprint.py

```diff
@@ -16,10 +29,187 @@ class FingerprintSubcommand(Subcommand):
+        rxn_cols=args.reaction_columns,
+        target_cols=None,
+        ignore_cols=None,
+        weight_col=None,
+        bounded=bounded,
+    )
+    featurization_kwargs = dict(
+        features_generators=args.features_generators, keep_h=args.keep_h, add_h=args.add_h
+    )
+
+    test_data = build_data_from_files(
+        args.test_path,
+        **format_kwargs,
+        p_descriptors=args.descriptors_path,
+        p_atom_feats=args.atom_features_path,
+        p_bond_feats=args.bond_features_path,
+        p_atom_descs=args.atom_descriptors_path,
+        **featurization_kwargs,
+    )
+    logger.info(f"test size: {len(test_data[0])}")
+    test_dsets = [
+        make_dataset(d, args.rxn_mode, args.multi_hot_atom_featurizer_mode) for d in test_data
+    ]
+
+    if multicomponent:
+        test_dset = data.MulticomponentDataset(test_dsets)
+    else:
+        test_dset = test_dsets[0]
+
+    X_d_scaler = input_scalers.get("X_d", None) if input_scalers else None
+    V_f_scaler = input_scalers.get("V_f", None) if input_scalers else None
+    E_f_scaler = input_scalers.get("E_f", None) if input_scalers else None
+    V_d_scaler = input_scalers.get("V_d", None) if input_scalers else None
+
+    if X_d_scaler is not None:
+        test_dset.normalize_inputs("X_d", X_d_scaler)
+    if V_f_scaler is not None:
+        test_dset.normalize_inputs("V_f", V_f_scaler)
+    if E_f_scaler is not None:
+        test_dset.normalize_inputs("E_f", E_f_scaler)
+    if V_d_scaler is not None:
+        test_dset.normalize_inputs("V_d", V_d_scaler)
+
+    test_loader = data.MolGraphDataLoader(
+        test_dset, args.batch_size, args.num_workers, shuffle=False
+    )
+
+    logger.info(model)
+
+    match RepresentationType.get(args.repr_type):
+        case RepresentationType.FINGERPRINT:
+            predictor = model.fingerprint
+            fingerprint_length = model.message_passing.output_dim
+        case RepresentationType.ENCODING:
+            predictor = model.encoding
+            fingerprint_length = model.predictor.ffn[-2][-1].out_features
+        case _:
+            raise RuntimeError("unreachable code reached!")
+
+    fingerprints = torch.Tensor()
+    for batch in test_loader:
+        with torch.no_grad():
+            if multicomponent:
+                fingerprints = torch.cat(
+                    (fingerprints, predictor(batch.bmgs, batch.V_ds, batch.X_d)), 0
+                )
+            else:
+                fingerprints = torch.cat(
+                    (fingerprints, predictor(batch.bmg, batch.V_d, batch.X_d)), 0
+                )
+
+    fingerprint_columns = [f"fp_{i}" for i in range(fingerprint_length)]
+    df_fingerprints = pd.DataFrame(fingerprints, columns=fingerprint_columns)
+    if output_path.suffix == ".pkl":
+        df_fingerprints = df_fingerprints.reset_index(drop=True)
+        df_fingerprints.to_pickle(output_path)
+    else:
+        df_fingerprints.to_csv(output_path, index=False)
```

Comment:
saving an array as a dataframe seems like a really bad serialization method. Why not just save as a binary file? It's more efficient in both I/O and disk space.

Reply:
I agree this isn't efficient, but I imagine many users might prefer the DF option for ease of use when they load the fingerprints later. We could also provide 2 output format options.

Reply:
I agree with the pros and cons of different file formats. I think it's a good idea to provide both options here.